ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml

```diff
@@ -0,0 +1,227 @@
+        }
+    }
+
+    component SectionDelegate: Rectangle {
+        id: sectionDelegate
+        height: 64
+        color: Theme.palette.statusListItem.backgroundColor
+
+        property bool isCollectible
+
+        RowLayout {
+            anchors.fill: parent
+            StatusFlatButton {
+                size: StatusBaseButton.Size.Small
+                icon.name: checked ? "chevron-down" : "next"
+                checked: sectionDelegate.isCollectible ? d.collectiblesExpanded : d.assetsExpanded
+                textColor: Theme.palette.baseColor1
+                textHoverColor: Theme.palette.directColor1
+                onToggled: {
+                    if (sectionDelegate.isCollectible)
+                        d.collectiblesExpanded = !d.collectiblesExpanded
+                    else
+                        d.assetsExpanded = !d.assetsExpanded
+                }
+            }
+            StatusBaseText {
+                Layout.fillWidth: true
+                text: sectionDelegate.isCollectible ? qsTr("Collectibles") : qsTr("Assets")
+                elide: Text.ElideRight
+            }
+        }
+    }
+
+    component Placeholder: ShapeRectangle {
+        property bool isCollectible
+        text: isCollectible ? qsTr("Your hidden collectibles will appear here") : qsTr("Your hidden assets will appear here")
+    }
+
+    Component {
+        id: tokenDelegate
+        ManageTokensDelegate {
+            isCollectible: model.isCollectible
+            controller: isCollectible ? root.collectiblesController : root.assetsController
+            dragParent: null
+            dragEnabled: false
+            keys: ["x-status-draggable-none"]
+            isHidden: true
+            // FIXME map source index of the item being hidden (because of the offset due to the ConcatModel)
+        }
+    }
+
+    Component {
+        id: tokenGroupDelegate
+        ManageTokensGroupDelegate {
+            isCollectible: model.isCollectible
+            controller: isCollectible ? root.collectiblesController : root.assetsController
+            dragParent: null
+            dragEnabled: false
+            keys: ["x-status-draggable-none"]
+            isHidden: true
+            // FIXME map source index of the item being hidden (because of the offset due to the ConcatModel)
+        }
+    }
+
+    contentItem: ColumnLayout {
+        spacing: 2 // subtle spacing for the dashed placeholders to be fully visible
+
+        ColumnLayout { // no assets placeholder
+            Layout.fillWidth: true
+            spacing: 0
+            visible: !d.assetsCount
+            SectionDelegate {
+                Layout.fillWidth: true
+            }
+            Placeholder {
+                Layout.fillWidth: true
+                visible: d.assetsExpanded
+            }
+        }
+
+        StatusListView {
+            Layout.fillWidth: true
+            Layout.preferredHeight: contentHeight
```

Comment:
We have one model here but if I understand correctly there is ListView unrolling causing everything instantiated up-front, at least when it's put in external scrollable and height of the control is not limited (the case in storybook).

Reply:
Yeah thanks, I'll check it out